packages/daemon/src/daemon.js

```diff
@@ -797,26 +868,33 @@ const makeDaemonCore = async (
       values: endowmentFormulaIdentifiers,
     };
     return /** @type {Promise<{ formulaIdentifier: string, value: unknown }>} */ (
-      provideValueForNumberedFormula(formula.type, formulaNumber, formula)
+      provideValueForNumberedFormula(formula.type, evalFormulaNumber, formula)
     );
   };
 
   /**
-   * @param {string} hubFormulaIdentifier
-   * A "naming hub" is an objected with a variadic lookup method. It includes
-   * objects such as guests and hosts.
-   * @param {string[]} petNamePath
-   * @returns {Promise<{ formulaIdentifier: string, value: unknown }>}
+   * Incarnates a `lookup` formula and synchronously adds it to the formula graph.
+   * The returned promise is resolved after the formula is persisted.
+   * @param {string} formulaNumber - The lookup formula's number.
+   * @param {string} hubFormulaIdentifier - The formula identifier of the naming
+   * hub to call `lookup` on. A "naming hub" is an objected with a variadic
+   * lookup method. It includes objects such as guests and hosts.
+   * @param {string[]} petNamePath - The pet name path to look up.
+   * @returns {Promise<{ formulaIdentifier: string, value: import('./types').EndoWorker }>}
    */
-  const incarnateLookup = async (hubFormulaIdentifier, petNamePath) => {
-    const formulaNumber = await randomHex512();
+  const incarnateNumberedLookup = (
```

Comment:
This will be tricky to replicate if we go on as I suggest to make lookup formulas single-hop.